File: Data Structures & Algorithms/validate-parentheses/submission-0.py
class Solution:
    def isValid(self, s: str) -> bool:
        st = []
        op = set(['(', '[', '{'])
        cl = set([')', ']', '}'])
        for sep in s:
            if sep in op:
                st.append(sep)
            elif sep in cl:
                if len(st) == 0:
                    return False
                match sep:
                    case ']':
                        if st[len(st)-1] != '[':
                            return False
                        else:
                            st.pop()
                    case ')':
                        if st[len(st)-1] != '(':
                            return False
                        else:
                            st.pop()
                    case '}':
                        if st[len(st)-1] != '{':
                            return False
                        else:
                            st.pop()
            else:
                return False

            print(st)
        return len(st) == 0

File: Data Structures & Algorithms/is-anagram/submission-0.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        unordered_map<char, int> data;
        if (s.size() != t.size()) { return false; }
        for (const auto& sep : s){
            data[sep]++;
        }
        for(const auto& sep : t){
            if (!data.count(sep)) {
                return false;
            } else {
                data[sep]--;
                if (data[sep] == 0) {
                    data.erase(sep);
                }
            }
        }
        return data.empty();
    }
};
